web/src/settings_components.ts

```diff
@@ -1525,3 +1525,46 @@ export function create_group_setting_widget({
 
     return pill_widget;
 }
+
+export function set_expires_on_text(
+    $elem: JQuery,
+    custom_input_time_value: number,
+    valid_to_text: string,
+): void {
+    if ($elem.val() === "custom") {
+        $elem.parent().find(".input-time-valid-till").hide();
+        set_custom_expires_on_text($elem, custom_input_time_value, valid_to_text);
+    } else {
+        $elem.parent().find(".input-time-valid-till").show();
+        $elem.parent().find(".input-time-valid-till").text(valid_to_text);
+    }
+}
+
+export function set_custom_expires_on_text(
+    $elem: JQuery,
+    custom_input_time_value: number,
+    valid_to_text: string,
+): void {
+    if (util.validate_custom_time_input(custom_input_time_value)) {
+        $elem.parent().find(".custom-input-time-valid-till").text(valid_to_text);
+    } else {
+        $elem
+            .parent()
+            .find(".custom-input-time-valid-till")
+            .text($t({defaultMessage: "Invalid custom time"}));
+    }
+}
+
+export function set_custom_time_inputs_visibility(
+    $elem: JQuery,
+    custom_input_time_unit: string,
+    custom_input_time_value: number,
+): void {
+    if ($elem.val() === "custom") {
+        $elem.parent().find(".custom-input-time-value").val(custom_input_time_value);
+        $elem.parent().find(".custom-input-time-unit").val(custom_input_time_unit);
+        $elem.parent().find(".custom-time-input-container").show();
+    } else {
+        $elem.parent().find(".custom-time-input-container").hide();
+    }
+}
```

Comment:
Pure code moves should always be done in a commit that makes no other changes; can you rework this PR to have a first commit doing that, maybe a second replacing the IDs with classes, and a third making the actual logic changes?

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.

Reply:
Also, these changes current send a string for the custom expiration date to the new shared `settings_components` function that's for invalid custom time inputs. I think the validation should remain in the invite module, which likely involves refactoring `invalid_to` to do that validation check.

Reply:
And as [noted in your other PR](https://github.com/zulip/zulip/pull/31957#issuecomment-2417280265) related to these changes, you'll need to attribute work that's been taken from other contributor's work. See https://zulip.readthedocs.io/en/latest/contributing/continuing-unfinished-work.html.